src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -291,69 +326,127 @@ private async Task<string> DownloadAsync(Uri zipUri)
-                    }
+                    UnsquashImage(filePath, scriptPath);
+                }
+                else
+                {
+                    MountSquashfsImage(filePath, scriptPath);
                 }
             }
-            else
+            else if (packageType == CodePackageType.Zip)
             {
-                using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationZipExtract))
+                // default to unzip for zip packages
+                if (_environment.IsMountEnabled())
+                {
+                    MountZipFile(filePath, scriptPath);
+                }
+                else
                 {
-                    _logger.LogInformation($"Extracting files to '{scriptPath}'");
-                    ZipFile.ExtractToDirectory(filePath, scriptPath, overwriteFiles: true);
-                    _logger.LogInformation($"Zip extraction complete");
+                    UnzipPackage(filePath, scriptPath);
                 }
             }
+        }
+
+        private CodePackageType GetPackageType(string filePath)
+        {
+            // Check file magic-number using `file` command.
+            (var output, _, _) = RunBashCommand($"file -b {filePath}", MetricEventNames.LinuxContainerSpecializationFileCommand);
+            if (output.StartsWith("Squashfs", StringComparison.OrdinalIgnoreCase))
+            {
+                return CodePackageType.Squashfs;
+            }
+            else if (output.StartsWith("Zip", StringComparison.OrdinalIgnoreCase))
+            {
+                return CodePackageType.Zip;
+            }
+            else if (FileIsAny(".squashfs", ".sfs", ".sqsh", ".img", ".fs"))
+            {
+                return CodePackageType.Squashfs;
+            }
+            else if (FileIsAny(".zip"))
+            {
+                return CodePackageType.Zip;
+            }
+            else
+            {
+                throw new InvalidOperationException($"Can't find CodePackageType to match {filePath}");
+            }
 
             bool FileIsAny(params string[] options)
                 => options.Any(o => filePath.EndsWith(o, StringComparison.OrdinalIgnoreCase));
         }
 
-        private void MountFsImage(string filePath, string scriptPath)
+        private void UnzipPackage(string filePath, string scriptPath)
+        {
+            using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationZipExtract))
+            {
+                _logger.LogInformation($"Extracting files to '{scriptPath}'");
+                ZipFile.ExtractToDirectory(filePath, scriptPath, overwriteFiles: true);
+                _logger.LogInformation($"Zip extraction complete");
+            }
+        }
+
+        private void UnsquashImage(string filePath, string scriptPath)
+            => RunBashCommand($"unsquashfs -f -d '{scriptPath}' '{filePath}'", MetricEventNames.LinuxContainerSpecializationUnsquash);
```

Comment:
Will error out `dir_scan: failed to make directory /home/site/wwwroot, because No such file or directory`.
Need to `mkdir -p {scriptPath}` before calling unsquashfs.